src/spatial_reaction_systems/spatial_reactions.jl

```diff
@@ -0,0 +1,96 @@
+    TransportReaction(Num(rate), args...)
+end
+# Creates a vector of TransportReactions.
+function transport_reactions(transport_reactions)
+    [TransportReaction(tr[1], tr[2]) for tr in transport_reactions]
+end
+
+# Macro for creating a transport reaction.
+macro transport_reaction(rateex::ExprValues, species::Symbol)
+    make_transport_reaction(MacroTools.striplines(rateex), species)
+end
+function make_transport_reaction(rateex, species)
+    parameters = [];
+    find_parameters_in_rate!(parameters, rateex)
+    quote
+        $(isempty(parameters) ? nothing : :(@parameters $(parameters...)))
+        @variables t
+        @species $(species)(t)
+        TransportReaction($rateex, $species)
+    end
+end
+
+# Gets the parameters in a transport reaction.
+ModelingToolkit.parameters(tr::TransportReaction) = convert(Vector{BasicSymbolic{Real}}, Symbolics.get_variables(tr.rate))
+
+# Gets the species in a transport reaction.
+species(tr::TransportReaction) = [tr.species]
+spatial_species(tr::TransportReaction) = [tr.species]
+
+# Checks that a transport reaction is valid for a given reaction system.
+function check_spatial_reaction_validity(rs::ReactionSystem, tr::TransportReaction; edge_parameters=[])
+    # Checks that the species exist in the reaction system (ODE simulation code becomes difficult if this is not required, as non-spatial jacobian and f function generated from rs is of wrong size).  
+    any(isequal(tr.species), species(rs)) || error("Currently, species used in TransportReactions must also be in non-spatial ReactionSystem. This is not the case for $(tr.species).")
+    # Checks that the rate does not depend on species.    
+    isempty(intersect(ModelingToolkit.getname.(species(rs)), ModelingToolkit.getname.(Symbolics.get_variables(tr.rate)))) || error("The following species were used in rates of a transport reactions: $(setdiff(ModelingToolkit.getname.(species(rs)), ModelingToolkit.getname.(Symbolics.get_variables(tr.rate)))).")
+    # Checks that the species does not exist in the system with different metadata.
+    any([isequal(tr.species, s) && !isequal(tr.species.metadata, s.metadata) for s in species(rs)]) && error("A transport reaction used a species, $(tr.species), with metadata not matching its lattice reaction system. Please fetch this species from the reaction system and used in transport reaction creation.")
+    any([isequal(rs_p, tr_p) && !equivalent_metadata(rs_p, tr_p) for rs_p in parameters(rs), tr_p in Symbolics.get_variables(tr.rate)]) && error("A transport reaction used a parameter with metadata not matching its lattice reaction system. Please fetch this parameter from the reaction system and used in transport reaction creation.")
+    # Checks that no edge parameter occur among rates of non-spatial reactions.
+    any([!isempty(intersect(Symbolics.get_variables(r.rate), edge_parameters)) for r in reactions(rs)]) && error("Edge paramter(s) were found as a rate of a non-spatial reaction.")
```

Comment:
This works as long as they are spatially varying. However, once a property is spatially varying, its value is a vector of length equal to the number of vertexes (for a non-spatial parameter) or the number of edges (for spatial parameters). Since these might be different, I wanted to keep the two sets distinct.

Does this make sense? If this kind of parameters aresomething you still think we should support, I can try and figure out if there are any good ways to make this a possibility.

Reply:
Hmm, maybe we need to use metadata to tag what is a edge vs. vertex parameter at construction time?

Also, are edge parameters directed for you? (So in a directed graph one can have two per edge?)

Reply:
Yes, edge parameters are directed. However, if an undirected graph is provided when the `LatticeReactionSystem` is created, this is recorded. Next, if an edge parameter have vector values (suggesting different values at each edge), and that vector have a length equal to the number of edges in the non-directed graph (or half to the number of edges in the directed one), this vector is expanded to generate identical parameter values in both directions.

Reply:
Maybe we could add a specific metadata tag for a parameter that have can be both? Then we could exempt it from this check, but throw an error if it's value is a vector?